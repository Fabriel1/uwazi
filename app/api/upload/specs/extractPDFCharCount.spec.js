import fs from 'fs';

import extracPDFCharCount from '../extractPDFCharCount';

describe('extractPDFCharCount', () => {
  fit('should extract number of characters per page', (done) => {
    const filepath = `${__dirname}/batman_wikipedia.pdf`;

    const fileData = new Uint8Array(fs.readFileSync(filepath));

    extracPDFCharCount(fileData)
    .then((pages) => {
      const pdfInfo = Object.keys(pages).map(key => pages[key].chars).sort((a, b) => a - b);
      expect(pdfInfo).toMatchSnapshot();
      done();
    })
    .catch(done.fail);
  });
});
