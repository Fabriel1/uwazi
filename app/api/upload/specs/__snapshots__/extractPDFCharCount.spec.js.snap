// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractPDFCharCount should extract number of characters per page 1`] = `
Array [
  3000,
  5306,
  9082,
  13060,
  16706,
  20821,
  25203,
  29576,
  32965,
  37123,
  41069,
  44835,
  47646,
  51293,
  54984,
  58444,
  62773,
  66968,
  71390,
  74972,
  78732,
  81977,
  85916,
  90009,
  94267,
  98232,
  103120,
  107698,
  109648,
]
`;
